src/JwtCredentialsManager.ts

```diff
@@ -65,21 +67,22 @@ export default abstract class JwtCredentialsManager<
       throw new Error('JwtCredentialManager: Unauthenticated');
     }
 
-    const credentials = await this.getCredentialsFromAuthorization(token);
+    this.credentialsPromise = Promise.resolve(
+      this.getCredentialsFromAuthorization(token),
+    );
+    await this.credentialsPromise;
 
     // Avoid race conditions with multiple updates.
     if (this.token !== token) {
       return;
     }
```

Comment:
```suggestion
    const credentialsPromise = Promise.resolve(
      this.getCredentialsFromAuthorization(token),
    );
    this.credentialsPromise = credentialsPromise;
    await credentialsPromise;

    // Don't schedule renewal if the credentials have been replaced.
    if (this.credentialsPromise !== credentialsPromise) {
      return;
    }
```

i think?

Reply:
Oh yeah scheduling renewal twice would be harmless since it will cancel the other one and always operate on the latest credentials